jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/jdbc/JdbcAdapter.java

```diff
@@ -1,273 +1,276 @@
 package com.exasol.adapter.jdbc;
 
+import java.sql.*;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
 import com.exasol.ExaConnectionInformation;
 import com.exasol.ExaMetadata;
-import com.exasol.adapter.AdapterException;
+import com.exasol.adapter.*;
 import com.exasol.adapter.capabilities.*;
 import com.exasol.adapter.dialects.*;
-import com.exasol.adapter.json.RequestJsonParser;
-import com.exasol.adapter.json.ResponseJsonSerializer;
-import com.exasol.adapter.metadata.*;
+import com.exasol.adapter.metadata.SchemaMetadata;
+import com.exasol.adapter.metadata.SchemaMetadataInfo;
 import com.exasol.adapter.request.*;
-import com.exasol.logging.CompactFormatter;
-import com.exasol.utils.JsonHelper;
-import com.exasol.utils.UdfUtils;
+import com.exasol.adapter.response.*;
 
-import java.io.OutputStream;
-import java.sql.*;
-import java.util.List;
-import java.util.Map;
-import java.util.logging.*;
-
-public class JdbcAdapter {
-    public static final int MAX_STRING_CHAR_LENGTH = 2000000;
-    private static Logger logger = null;
+public class JdbcAdapter implements VirtualSchemaAdapter {
+    private static final String SCALAR_FUNCTION_PREFIX = "FN_";
+    private static final String PREDICATE_PREFIX = "FN_PRED_";
+    private static final String AGGREGATE_FUNCTION_PREFIX = "FN_AGG_";
+    private static final String LITERAL_PREFIX = "LITERAL_";
+    private static final Logger LOGGER = Logger.getLogger(JdbcAdapter.class.getName());
 
     /**
-     * This method gets called by the database during interactions with the virtual
-     * schema.
+     * This method gets called by the database during interactions with the virtual schema.
      *
-     * @param meta  Metadata object
-     * @param input JSON request, as defined in the Adapter Script API
+     * @param metadata   Metadata object
+     * @param rawRequest JSON request, as defined in the Adapter Script API
      * @return JSON response, as defined in the Adapter Script API
+     * @throws AdapterException in case the request is not recognized
+     * @deprecated As of Virtual Schema version 1.8.0 you should use
+     *             {@link com.exasol.adapter.RequestDispatcher#adapterCall(ExaMetadata, String)} as entry point instead.
      */
-    public static String adapterCall(final ExaMetadata meta, final String input) throws Exception {
-        String result = "";
+    @Deprecated
+    public static String adapterCall(final ExaMetadata metadata, final String rawRequest) throws AdapterException {
+        final VirtualSchemaAdapter adapter = new JdbcAdapter();
+        registerAdapterForSqlDialect(adapter, "DB2"); // FIXME: replace this hard-coded registration
+        registerAdapterForSqlDialect(adapter, "EXASOL"); // FIXME: replace this hard-coded registration
+        registerAdapterForSqlDialect(adapter, "GENERIC"); // FIXME: replace this hard-coded registration
```

Comment:
The HIVE dialect is not yet registered here.

Reply:
Fixed.